pkg/model/resource/resource.go

```diff
@@ -19,6 +19,8 @@ package resource
 import (
 	"fmt"
 	"strings"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/internal/validation"
```

Comment:
consumers cannot use the internal one. Should it be really internal?

Reply:
I didn't touch that, it was internal previously. I think the reason to make it internal is that it was borrowed code from another repo that we didn't wan't to add as a dependency.